packages/xstate-fsm/test/types.test.ts

```diff
@@ -58,4 +59,31 @@ describe('matches', () => {
       ((_accept: string) => {})(state.context.count);
     }
   });
+
+  it('should require actions on wildcard transitions to handle all event types', () => {
+    type Context = {};
+    type FooEvent = { type: 'foo'; foo: string };
+    type BarEvent = { type: 'bar'; bar: number };
+    type Event = FooEvent | BarEvent;
+    type State = { value: 'one'; context: Context };
+    createMachine<Context, Event, State>({
+      context: {},
+      initial: 'one',
+      states: {
+        one: {
+          on: {
+            foo: {
+              target: 'one',
+              actions: (_context: Context, _event: InitEvent | FooEvent) => {}
+            },
+            // @ts-expect-error
```

Comment:
to accompany [the other suggested change](https://github.com/statelyai/xstate/pull/4065/files#r1230935353) I'd change this to smth like:
```suggestion
            // @x-ts-expect-error
```

We'll "turn this on" at later point in time.